tutorials/getting_started/so_you_built_a_model.md

```diff
@@ -0,0 +1,182 @@
+
+Here we have one additional step: we need to associate our custom predictor with our model.
+This is accomplished via the
+
+```python
+         predictor_overrides: Dict[str, str] = {},
+```
+
+parameter, which is a mapping from `Model` name to `Predictor` name.
+
+In our case we'd have to call
+
+```python
+    main(prog="python run.py",
+         predictor_overrides={'bidaf': 'machine-comprehension'})
+```
+
+to specify that when we see a `bidaf` model and need a `Predictor`
+we should use the `machine-comprehension` one.
+(As mentioned before, these are already associated
+[by default](https://github.com/allenai/allennlp/blob/master/allennlp/commands/predict.py#L35),
+but for tutorial reasons we're pretending they're not.)
+
+At this point you just need an input file in JSON lines format:
+
+```
+{"question": "First question...", "passage": "First passage..."}
+{"question": "Second question...", "passage": "Second passage..."}
+...
+```
+
+If you run your script, you should get a help message:
+
+```bash
+$ run.py predict
+usage: python -m allennlp.run predict [-h]
+                                      [--output-file OUTPUT_FILE]
+                                      [--batch-size BATCH_SIZE]
+                                      [--silent]
+                                      [--cuda-device CUDA_DEVICE]
+                                      [-o OVERRIDES]
+                                      archive_file input_file
+```
+
+In the simplest case, just specify the `archive_file`
+(that's the trained model we downloaded above)
+and the `input_file`
+(that's the JSON lines file above)
+and your predictions will get printed to the console.
+If you'd rather have them in a file, specify the `--output-file` parameter.
+
+It's also possible to use non-JSON formats, for example, if you had CSV data.
+In this case your `Predictor` would need to override `load_line` and `dump_line`.
+
+## Running a Web Demo
+
+Once you have a trained model and a predictor,
+it's easy to run a simple web demo
+using `allennlp.service.server_simple.make_app()`:
+
+```python
+from allennlp.models.archival import load_archive
+from allennlp.service.predictors import Predictor
+from allennlp.service.server_simple import make_app
+
+from gevent.wsgi import WSGIServer
+
+archive = load_archive('/path/to/trained/bidaf/model.tar.gz')
+predictor = Predictor.from_archive(archive, 'machine-comprehension')
+
+app = make_app(predictor=predictor,
+               field_names=['passage', 'question'])
+
+http_server = WSGIServer(('0.0.0.0', 8888), app)
+http_server.serve_forever()
+```
+
+The important things to notice are
+
+* We had to explicitly specify which predictor to use. If you were
+  using a custom one, you'd need to import it at the top of the script.
+* We had to explicitly specify the field names. These are used to generate
+  the web interface but also (more importantly) as the keys in the input JSON,
```

Comment:
This sentence wasn't clear to me the first time I read it.  Maybe it's because you have "used to generate" coordinated with "[used] as the keys", which is a pretty rare construct.  A simpler sentence structure here would probably make this more clear.  Maybe even just repeating "are used" before "as the keys" would be enough...